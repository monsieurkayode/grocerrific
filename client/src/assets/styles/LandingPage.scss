#landing {
  @extend %flex__container_column;
  @extend %nowrap;
  min-height: 100%;

  header { @include header(); }

  section {
    flex: 1;
    @extend %flex__container_row;
    width: 100%;
    margin-top: 10%;

    .section__left {
      width: 48%;
      margin-right: px-to-rem(20px);

      .info__wrapper_top,
      .info__wrapper_bottom {
        color: #FFF;
        margin-left: 15%;
      }

      .info__wrapper_top {
        line-height: px-to-rem(80px);
  
        .info__wrapper_text {
          font-family: $font-family-prompt;
          letter-spacing: -2px;
          font-size: px-to-rem(90px);
        }
      }

      .info__wrapper_bottom {
        border-left: 3px solid #58d4f8;
        margin-top: px-to-rem(40px);
        width: 80%;
  
        .info__wrapper_text {
          padding-left: px-to-rem(40px);
          font-family: $font-family-josefin;
          font-weight: 200;
          font-size: px-to-rem(18px);
          line-height: px-to-rem(30px);
        }
      }

      .buttons__wrapper {
        @extend %flex__container_row;
        margin-top: px-to-rem(60px);
        user-select: none;
        margin-left: 15%;

        .info {
          display: inline-block;
          font-family: $font-family-prompt;
          letter-spacing: 1px;

          .info__top {
            font-size: px-to-rem(10px);

            span{
              font-size: px-to-rem(8px);
              font-weight: 100;
            }
          }
          .info__bottom {
            font-size: px-to-rem(16px);
            font-weight: 700;
          }
        }

        .designed__by {
          margin-right: px-to-rem(30px);
          display: inline-block;
          width: 30%;
          min-width: px-to-rem(185px);
          background: #ffc741;
          border-radius: 8px;
          color: #FFF;
          padding: px-to-rem(20px);
          cursor: pointer;

          img {
            display: inline-block;
            margin-right: px-to-rem(10px);
          }
        }

        .join {
          display: inline-block;
          width: 40%;
          min-width: px-to-rem(245px);
          background: #f6f6f6;
          border-radius: 8px;
          padding: px-to-rem(20px) px-to-rem(30px);
          cursor: pointer;

          img {
            display: inline-block;
            margin-right: px-to-rem(10px)
          }

          .info {
            color: #bb8fde;
          }
        }
      }
    }

    .section__right {
      width: 48%;

      @keyframes bounce {
        0% { transform: translateY(-10px) }
        100% { transform: translateY(10px) }
      }

      #card {
        max-width: px-to-rem(600px);
        min-width: px-to-rem(300px);
        min-height: px-to-rem(400px);
        background: #FFF url('../images/groceries.jpg') no-repeat center center / contain;
        border-radius: 5px;
        box-shadow: 0 0 30px 0 $light-grey-1,
                    0 15px 20px -6px $light-grey-1;
        animation: bounce;
        animation-duration: 3s;
        animation-iteration-count: infinite;
        animation-direction: alternate-reverse;
      }
    }
  }
}

