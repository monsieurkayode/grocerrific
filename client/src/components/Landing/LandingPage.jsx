import React from 'react';
import { Link } from 'react-router-dom';

import Header from '../common/Header';

import cart from '../../assets/images/cart.png';
import brush from '../../assets/images/brush.png';

const LandingPage = () => (
  <main id="landing" className="container">
    <Header>
      <div className="header__right">
        <Link to="/store">
          <div className="start text__medium_bold uppercase">
            Enter Store
          </div>
        </Link>
      </div>
    </Header>
    <section>
      <div className="section__left">
        <div className="info__wrapper_top">
          <div className="info__wrapper_text text__bold">Grocerrific</div>
          <div className="info__wrapper_text text__bold">Marketplace</div>
        </div>
        <div className="info__wrapper_bottom">
          <p className="info__wrapper_text">
            We deliver on our promise of giving our users the best grocery
            shopping experience. Because we care a lot about our planet, all our
            products are organic and did we forget to mention terrific too?
          </p>
        </div>
        <div className="buttons__wrapper">
          <div className="designed__by">
            <img src={brush} alt="Brush" />
            <div className="info uppercase">
              <p className="info__top">Designed By</p>
              <p className="info__bottom">Emperor<span>&trade;</span></p>
            </div>
          </div>
          <div className="join">
            <img src={cart} alt="Cart" />
            <div className="info uppercase">
              <p className="info__top">Join the</p>
              <p className="info__bottom">Marketplace</p>
            </div>
          </div>
        </div>
      </div>
      <div className="section__right">
        <div id="card" />
      </div>
    </section>
  </main>
);

export default LandingPage;
