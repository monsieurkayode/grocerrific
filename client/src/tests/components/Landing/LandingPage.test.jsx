import React from 'react';
import { shallow } from 'enzyme';
import LandingPage from '../../../components/Landing/LandingPage';

describe('LandingPage Component', () => {
  const wrapper = shallow(<LandingPage />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
