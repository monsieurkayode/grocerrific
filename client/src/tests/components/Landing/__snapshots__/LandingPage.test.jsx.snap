// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component renders correctly 1`] = `
<main
  className="container"
  id="landing"
>
  <withRouter(Header)>
    <div
      className="header__right"
    >
      <Link
        replace={false}
        to="/store"
      >
        <div
          className="start text__medium_bold uppercase"
        >
          Enter Store
        </div>
      </Link>
    </div>
  </withRouter(Header)>
  <section>
    <div
      className="section__left"
    >
      <div
        className="info__wrapper_top"
      >
        <div
          className="info__wrapper_text text__bold"
        >
          Grocerrific
        </div>
        <div
          className="info__wrapper_text text__bold"
        >
          Marketplace
        </div>
      </div>
      <div
        className="info__wrapper_bottom"
      >
        <p
          className="info__wrapper_text"
        >
          We deliver on our promise of giving our users the best grocery shopping experience. Because we care a lot about our planet, all our products are organic and did we forget to mention terrific too?
        </p>
      </div>
      <div
        className="buttons__wrapper"
      >
        <div
          className="designed__by"
        >
          <img
            alt="Brush"
            src="test-file-mock"
          />
          <div
            className="info uppercase"
          >
            <p
              className="info__top"
            >
              Designed By
            </p>
            <p
              className="info__bottom"
            >
              Emperor
              <span>
                â„¢
              </span>
            </p>
          </div>
        </div>
        <div
          className="join"
        >
          <img
            alt="Cart"
            src="test-file-mock"
          />
          <div
            className="info uppercase"
          >
            <p
              className="info__top"
            >
              Join the
            </p>
            <p
              className="info__bottom"
            >
              Marketplace
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="section__right"
    >
      <div
        id="card"
      />
    </div>
  </section>
</main>
`;
